agent/api/task/task.go

```diff
@@ -953,6 +1014,91 @@ func (task *Task) generateENIExtraHosts() []string {
 	return extraHosts
 }
 
+// shouldOverridePIDMode returns true if the PIDMode of the container needs
+// to be overridden. It also returns the override string in this case. It returns
+// false otherwise
+func (task *Task) shouldOverridePIDMode(container *apicontainer.Container, dockerContainerMap map[string]*apicontainer.DockerContainer) (bool, string) {
+	// If the container is an internal container (ContainerEmptyHostVolume,
+	// ContainerCNIPause, or ContainerNamespacePause), then PID namespace for
+	// the container itself should be private (default Docker option)
+	if container.IsInternal() {
+		return false, ""
+	}
+
+	if task.getPIDMode() == pidModeHost {
+		return true, pidModeHost
+	} else if task.getPIDMode() == pidModeTask {
+		pauseContName, ok := task.findPauseContainerName(apicontainer.ContainerNamespacePause)
+		if !ok {
+			seelog.Critical("Namespace Pause container not found in the task: %s", task.String())
+			return false, ""
+		}
+		pauseToGiveDocker, ok := dockerContainerMap[pauseContName]
```

Comment:
what does `pauseToGiveDocker` mean?

Reply:
This is the Docker container name that is contained in the map. We will 'give' this container to the Docker api for the --pid or --ipc flag

Reply:
Got it. it's a little confusing. I would prefer just calling it pauseContainerID or something similar.

Reply:
Changed!